packages/ui-components/src/components/layout/Dialog.vue

```diff
@@ -86,7 +86,12 @@
               >
                 <XMarkIcon class="h-6 w-6" />
               </FormButton>
-              <div ref="slotContainer" :class="slotContainerClasses" @scroll="onScroll">
+              <div
+                ref="slotContainer"
+                v-memo="isClosing ? [memoKey] : [(memoKey = Math.random())]"
```

Comment:
Made it so that when the dialog is starting to close and the fade-out transition starts playing, the dialog's contents stop being reactive and show the same DOM that was shown when closing was initiated.

This resolves the issue when you have a single value from which you derive both the dialog's state and also whether it's open or not. It used to be that in such cases when you'd close the dialog, the data inside the dialog would disappear too and you could see it in the closing transition.

Reply:
```
/**
 * This allows you to only mutate modelToEdit, and editDialogOpen will update automatically depending on whether
 * you've chosen a model to edit.
 *
 * But up until now, closing editDialogOpen would also wipe modelToEdit and make the dialog show up as empty for the split second while its closing
 */
const modelToEdit = ref(null)
const editDialogOpen = computed({
  get: () => !!modelToEdit.value,
  set: (newVal) => {
    if (!newVal) {
      modelToEdit.value = null
    }
  }
})

```